File: azkry_android/app/src/main/kotlin/com/azkry/app/features/friday/models/FridaySunnah.kt
package com.azkry.app.features.friday.models

import androidx.annotation.StringRes
import com.azkry.app.R
import com.azkry.app.core.models.WorshipTask

/**
 * The Friday sunan checklist. [Kahf] deliberately maps onto the tracking
 * task's key so checking it here also counts in worship tracking; the others
 * use their own keys, which the scoring ignores.
 */
enum class FridaySunnah(
    @param:StringRes val titleRes: Int,
    val taskKey: String,
) {
    Ghusl(R.string.friday_ghusl, "FridayGhusl"),
    Perfume(R.string.friday_perfume, "FridayPerfume"),
    BestClothes(R.string.friday_best_clothes, "FridayBestClothes"),
    EarlyMosque(R.string.friday_early_mosque, "FridayEarlyMosque"),
    Dua(R.string.friday_dua, "FridayDua"),
    Kahf(R.string.friday_kahf, WorshipTask.Kahf.key),
}

File: azkary_ios/Azkry/Features/Friday/Models/FridaySunnah.swift
enum FridaySunnah: String, CaseIterable, Identifiable, Sendable {
    case ghusl
    case perfume
    case bestClothes
    case earlyMosque
    case dua
    case kahf

    var id: String { rawValue }

    var title: String {
        switch self {
        case .ghusl: "الاغتسال"
        case .perfume: "التطيّب"
        case .bestClothes: "أحسن الثياب"
        case .earlyMosque: "التبكير للمسجد"
        case .dua: "تحرّي ساعة الإجابة"
        case .kahf: "قراءة سورة الكهف"
        }
    }
}
